python/lsst/ctrl/mpexec/cmdLineFwk.py

```diff
@@ -399,6 +399,10 @@ def runPipeline(self, graph, taskFactory, args):
         if not args.init_only:
             executor = MPGraphExecutor(numProc=args.processes, timeout=self.MP_TIMEOUT)
             executor.execute(graph, butler, taskFactory)
+        # Explicitly shutdown the butler as automatic collection can
+        # sometimes lead to an exception being thrown on exit. This
+        # does not affect processing but is not friendly to the user.
+        butler.shutdown()
```

Comment:
Moving this call to butler instead of registry is certainly better if we can't get rid of it completely. Would it better to call it `close()` instead of `shutdown()`? We could then use `contextlib.closing` for context handling as Tim proposed.